zulipterminal/ui_tools/boxes.py

```diff
@@ -592,32 +604,37 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
                     if header.focus_col == self.FOCUS_HEADER_BOX_STREAM:
                         stream_name = (header[self.FOCUS_HEADER_BOX_STREAM]
                                        .edit_text)
-                        if not self.model.is_valid_stream(stream_name):
+                        if (not self.model.is_valid_stream(stream_name)
+                                and self.loosing_focus_from_stream_box(key)):
                             self.footer_notify_invalid_recipient(
                                 'Invalid stream name.')
                             return key
-                        user_ids = self.model.get_other_subscribers_in_stream(
-                                                    stream_name=stream_name)
-                        self.recipient_user_ids = user_ids
-                        self.stream_id = self.model.stream_id_from_name(
-                                                                stream_name)
+                        if not is_primary_direction_key(key):
```

Comment:
In place of recurring conditionals, we could just check for it once and return early? That would remove all the following code changes.

Reply:
All of these conditionals have their corresponding `else`/`elif` clauses, so returning early might not give the expected behavior. Let me know if I misunderstood you. :)

Reply:
Right, we would have to return early from three places.

I think a cleaner way would be to provision another `elif` and check the stream validity separately.